src/package.coffee

```diff
@@ -200,7 +200,13 @@ class Package
 
   activateResources: ->
     @activationDisposables = new CompositeDisposable
-    @activationDisposables.add(atom.keymaps.add(keymapPath, map)) for [keymapPath, map] in @keymaps
+
+    @activationDisposables.add atom.config.observe "core.disabledKeymaps", {}, (map) =>
+      value = not _.include(map ? [], @name)
+      if value
+        @activateKeymaps()
+      else if not value
+        @deactivateKeymaps()
```

Comment:
> Would you suggest splitting it into onDidChange for the observer and an initial config check for activation?

I don't have a strong opinion on that; go with whichever way makes the most sense as you're writing the code.